src/main/java/org/tron/keystore/Wallet.java

```diff
@@ -0,0 +1,239 @@
+package org.tron.keystore;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.UUID;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.spongycastle.crypto.digests.SHA256Digest;
+import org.spongycastle.crypto.generators.PKCS5S2ParametersGenerator;
+import org.spongycastle.crypto.generators.SCrypt;
+import org.spongycastle.crypto.params.KeyParameter;
+import org.tron.common.crypto.ECKey;
+import org.tron.common.crypto.Hash;
+import org.tron.common.utils.ByteArray;
+
+/**
+ * <p>Ethereum wallet file management. For reference, refer to <a href="https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition">
+ * Web3 Secret Storage Definition</a> or the <a href="https://github.com/ethereum/go-ethereum/blob/master/accounts/key_store_passphrase.go">
+ * Go Ethereum client implementation</a>.</p>
+ *
+ * <p><strong>Note:</strong> the Bouncy Castle Scrypt implementation {@link SCrypt}, fails to comply
+ * with the following Ethereum reference <a href="https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition#scrypt">
+ * Scrypt test vector</a>:</p>
+ *
+ * <pre>
+ * {@code
+ * // Only value of r that cost (as an int) could be exceeded for is 1
+ * if (r == 1 && N_STANDARD > 65536)
+ * {
+ *     throw new IllegalArgumentException("Cost parameter N_STANDARD must be > 1 and < 65536.");
+ * }
+ * }
+ * </pre>
+ */
+public class Wallet {
+
+  private static final int N_LIGHT = 1 << 12;
+  private static final int P_LIGHT = 6;
+
+  private static final int N_STANDARD = 1 << 18;
+  private static final int P_STANDARD = 1;
+
+  private static final int R = 8;
+  private static final int DKLEN = 32;
+
+  private static final int CURRENT_VERSION = 3;
+
+  private static final String CIPHER = "aes-128-ctr";
+  static final String AES_128_CTR = "pbkdf2";
+  static final String SCRYPT = "scrypt";
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Explicitly declare the visibility for "SCRYPT". ([squid:S2039](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2039))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=938&issues=AWPK23ifkDisRwWBfuxM&open=AWPK23ifkDisRwWBfuxM)

[ISSUE_KEY]: AWPK23ifkDisRwWBfuxM